src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -735,9 +744,9 @@ internal async Task SendInvocationRequest(ScriptInvocationContext context)
                     context.CancellationToken.Register(() => SendInvocationCancel(invocationRequest.InvocationId));
                 }
 
-                if (!string.IsNullOrEmpty(_workerCapabilities.GetCapabilityState(RpcWorkerConstants.EnableHttpProxying)) && context.FunctionMetadata.IsHttpTriggerFunction())
+                if (!string.IsNullOrEmpty(_workerCapabilities.GetCapabilityState(RpcWorkerConstants.HttpUri)) && context.FunctionMetadata.IsHttpTriggerFunction())
```

Comment:
We can optimize this by caching the results of the HTTP capability evaluation (when we receive the capabilities) to avoid a lookup with every invocation.